### PR TITLE
Use jetpack compose for buttons in task fragments

### LINKED ISSUE #1795: Migrate to Jetpack Compose
Some early experimentation in #1629

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #1795

This PR replaces xml layouts for action buttons with jetpack compose.

[compose-task-button.webm](https://github.com/google/ground-android/assets/8918023/ac841e8a-f91c-4ab9-84cb-e62fbb9ed5c2)

@gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt [hunk 1]
```diff
@@ -192,21 +198,30 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
...
+  /** Adds the action buttons to the UI. */
+  private fun renderButtons() {
+    taskView.actionButtonsContainer.composeView.apply {
+      setContent {
+        AppTheme {
+          Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier.fillMaxWidth().padding(8.dp),
+          ) {
+            // TODO: Previous button should always be positioned to the left of the screen. Rest
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt:216]: Please add tracking bug.

**Review by gino-m — CHANGES_REQUESTED** [review]
Overall looks good! @sufyanAbbasi Do you think this is safe to merge, or do we want to implement e2e testing first?

**Comment by shobhitagarwal1612** [comment]
The tests are passing locally. Not sure what the problem is with remote builds.

**Comment by gino-m** [comment]
> The tests are passing locally. Not sure what the problem is with remote builds.

The "Update build status" step failed, not the actual tests. Flake?

**Comment by shobhitagarwal1612** [comment]
We upload the results in this step and then fail the build. So this is WAI

**Comment by gino-m** [comment]
> We upload the results in this step and then fail the build. So this is WAI

I see. So the actual error is in the unit tests:

```
com.google.android.ground.ui.signin.SignInFragmentTest > Back press should finish activity FAILED
    kotlinx.coroutines.test.UncaughtExceptionsBeforeTest at TestScope.kt:242
```

It seems the activity isn't closing on "Back" on GCB for some reason?

**Comment by shobhitagarwal1612** [comment]
I can see this test failing occasionally on local builds as well. Should we retry?

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt]: Done

**Review by gino-m — APPROVED** [review]
⏺️

**Comment by shobhitagarwal1612** [comment]
Even before this change, the buttons were generated dynamically without a resource ID. So we can use text or content description (for image buttons)

**Pull request merged**